src/components/Modals/DailyLimitModal/DailyLimitModal.tsx

```diff
@@ -14,84 +13,101 @@ type TDailyLimitModalProps = {
                     </Button>
-                    <Button onClick={() => mutate({ upgrade_limits: 1 })} size='lg' textSize='sm'>
+                    <Button onClick={() => mutate({ upgrade_limits: 1 })} size='lg' textSize={textSize}>
                         <Localize i18n_default_text='Yes, continue' />
                     </Button>
-                </div>
-            </>
-        );
+                </>
+            ),
+            header: <Localize i18n_default_text='Are you sure?' />,
+        },
+        error: {
+            body: (
+                <Localize i18n_default_text='Sorry, we’re unable to increase your limits right now. Please try again in a few minutes.' />
+            ),
+            footer: (
+                <Button onClick={onRequestClose} size='lg' textSize={textSize}>
+                    <Localize i18n_default_text='Ok' />
+                </Button>
+            ),
+            header: <Localize i18n_default_text='An internal error occurred' />,
+        },
+        loading: {
+            body: <Loader isFullScreen={false} />,
+            footer: null,
+            header: null,
+        },
+        success: {
+            body: (
+                <Localize
+                    i18n_default_text='Your daily limits have been increased to {{daily_buy_limit}} {{currency}} (buy) and {{daily_sell_limit}} {{currency}} (sell).'
+                    values={{ currency, daily_buy_limit, daily_sell_limit }}
+                />
+            ),
+            footer: (
+                <Button onClick={onRequestClose} size='lg' textSize={textSize}>
+                    <Localize i18n_default_text='Ok' />
+                </Button>
+            ),
+            header: <Localize i18n_default_text='Success!' />,
+        },
+    };
+
+    const getModalContent = () => {
+        if (isLoading) {
+            return modalContent.loading;
+        } else if (isSuccess) {
+            return modalContent.success;
+        } else if (error) {
+            return modalContent.error;
+        }
+        return modalContent.default;
     };
 
     return (
-        // TODO: below modal will be rewritten to use @deriv/ui modal
         <Modal
             ariaHideApp={false}
             className='daily-limit-modal'
             isOpen={isModalOpen}
             onRequestClose={onRequestClose}
-            style={customStyles}
+            shouldCloseOnOverlayClick={false}
             testId='dt_daily_limit_modal'
         >
-            {getModalContent()}
+            {!isLoading && (
+                <Modal.Header hideBorder hideCloseIcon>
+                    <Text color='prominent' size={headerTextSize} weight='bold'>
+                        {getModalContent().header}
+                    </Text>
+                </Modal.Header>
+            )}
+            <Modal.Body className={clsx('py-[0.8rem] px-[2.4rem]', { 'mx-auto py-[2.4rem]': isLoading })}>
+                <Text size={textSize}>{getModalContent().body}</Text>
```

Comment:
Why are  we using content like this? Can't we just use reusable component?

Reply:
this would require repetition of modal header and also text component.